File: app/src/main/java/com/szymkowski/personaltrainercompanion/payments/domain/PaymentMapper.java
package com.szymkowski.personaltrainercompanion.payments.domain;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.Mappings;
import org.mapstruct.ReportingPolicy;
import org.mapstruct.factory.Mappers;


@Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface PaymentMapper {

    PaymentMapper INSTANCE = Mappers.getMapper(PaymentMapper.class);

    @Mappings({
            @Mapping(source = "paymentDate", target = "paymentDate"),
            @Mapping(source = "numberOfClassesPaid", target = "numberOfClassesPaid")
    })
    PaymentDTO paymentToPaymentDTO(Payment payment);

    @Mappings({
            @Mapping(source = "paymentDate", target = "paymentDate"),
            @Mapping(source = "numberOfClassesPaid", target = "numberOfClassesPaid")
    })
    Payment paymentDTOToPayment(PaymentDTO paymentDTO);

}

File: app/src/main/java/com/szymkowski/personaltrainercompanion/payments/domain/PaymentMapper.kt
package com.szymkowski.personaltrainercompanion.payments.domain



object PaymentMapper {

    fun paymentToPaymentDTO(payment: Payment): PaymentDTO {
        val (id, date, number) = payment
        return PaymentDTO(id, date, number)
    }

    fun paymentDTOToPayment(paymentDTO: PaymentDTO): Payment {
        val (id, date, number) = paymentDTO
        return Payment(id, date, number)
    }

}
